docs/book/src/topics/node-outbound-lb.md

```diff
@@ -0,0 +1,64 @@
+# Node Outbound Load Balancer
+
+This document describes how to configure your clusters' node outbound load balancer.
+
+### Public Clusters
+
+For public clusters ie. clusters with api server load balancer type set to `Public`, CAPZ automatically configures a node outbound load balancer with the default settings.
+
+To provider custom settings for the node outbound load balacer, use the `nodeOutboundLB` section in cluster configuration.
+
+Here is an example of a node outbound load balancer with `frontendIPsCount` set to 3. CAPZ will read this value and create 3 front end ips for this load balancer.
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+kind: AzureCluster
+metadata:
+  name: my-public-cluster
+  namespace: default
+spec:
+  location: eastus
+  networkSpec:
+    apiServerLB:
+      type: Public
+    nodeOutboundLB:
+      frontendIPsCount: 3
+```
+
+<aside class="note warning">
+
+<h1> Warning </h1>
+
+Only `frontendIPsCount` is allowed to be configured for any node outbound load balancer. Trying to modify any other value will result in a validation error.
+
+</aside>
+
+### Private Clusters
+
+For private clusters ie. clusters with api server load balancer type set to `Internal`, CAPZ does not create a node outbound load balancer by default. 
```

Comment:
> For private clusters ie. clusters with api server load balancer type set to `Internal`, CAPZ does not create a node outbound load balancer by default

I don't think that's true. CAPZ always creates a node outbound LB to give outbound access to the worker nodes. For private clusters, it also creates a control plane outbound LB to give outbound access to the control plane nodes (in non-private clusters the API Server LB is public so it already gives outbound to the control planes).

Reply:
IIUC, this pr changes the default behavior because we now want the ability to disable node outbound LB for private clusters by not including `nodeOutboundLB` section. So, unless the user explicitly sets nodeOutboundLB, capz will not create a node outbound lb. That's what I've tried to explain here. wdyt?

Reply:
Got it, I didn't realize you were including that change in this PR. In that case it also partly addresses #1159 right?

Reply:
Yes, partially. I'll update the pr description to include this change.